docs/LoongArch-ELF-ABI-CN.adoc

```diff
@@ -647,6 +647,203 @@ v1.00
 |R_LARCH_GNU_VTENTRY
 |GNU C++ vtable 支持
 |
+
+4+|... 保留项
+
+|64
+|R_LARCH_B16
+|18 位相对 PC 跳转
+|`+(*(uint32_t *) PC) [25 ... 10] = (S+A-PC) [17 ... 2]+`
+
+带 18 位有符号数溢出和4字节对齐检测功能
+
+|65
+|R_LARCH_B21
+|23 位相对 PC 跳转
+|`+(*(uint32_t *) PC) [4 ... 0] = (S+A-PC) [22 ... 18],+`
+
+`+(*(uint32_t *) PC) [25 ... 10] = (S+A-PC) [17 ... 2]+`
+
+带 23 位有符号数溢出和4字节对齐检测功能
+
+|66
+|R_LARCH_B26
+|28 位相对 PC 跳转
+|`+(*(uint32_t *) PC) [9 ... 0] = (S+A-PC) [27 ... 18],+`
+
+`+(*(uint32_t *) PC) [25 ... 10] = (S+A-PC) [17 ... 2]+`
+
+带 28 位有符号数溢出和4字节对齐检测功能
+
+|67
+|R_LARCH_ABS_HI20
+|32 位绝对地址的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (S+A) [31 ... 12]+`
+
+|68
+|R_LARCH_ABS_LO12
+|32 位绝对地址的[0:11]位
+|`(*(uint32_t *) PC) [21 ... 10] = (S+A) [11 ... 0]+`
+
+|69
+|R_LARCH_ABS64_LO20
+|64 位绝对地址[32:51]位
+|`(*(uint32_t *) PC) [24 ... 5] = (S+A) [51 ... 32]+`
+
+|70
+|R_LARCH_ABS64_HI12
+|64 位绝对地址[52:63]位
+|`(*(uint32_t *) PC) [21 ... 10] = (S+A) [63 ... 52]+`
+
+|71
+|R_LARCH_PCALA_HI20
+|32位相对PC偏移的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (((S+A) & ~0xfff) - (PC & ~0xfff)) [31 ... 12]+`
+
+|72
+|R_LARCH_PCALA_LO12
+|32位相对PC偏移的[0:11]位
+|`(*(uint32_t *) PC) [21 ... 10] = (S+A) [11 ... 0]+`
```

Comment:
The text description is completely different than the symbolic definition.  The symbolic description is exactly same as `R_LARCH_ABS_LO12`.

Reply:
I guess you want to use something like:

```
pcalau12i $t0, %pcala_hi20(something)
ld.d $t0, $t0, %lo12(something)
```

to load `something`, if I interpret @SixWeining's comment in https://reviews.llvm.org/D128427 correctly.  Then, (1) you don't need a separate "%pcala_lo12" (it's same as "%abs_lo12" so only one "%lo12" is enough); (2) you need to alter the definition of %pcala_hi20 to account the carry from %lo12 into consideration.  Like:

`R_LARCH_PCALA_HI20` ::= `(*(u32 *)PC) [24..5] = (((S+A+0x800) & ~0xfff) - (PC & ~0xfff))[31..12]`

Reply:
R_LARCH_ABS_LO12 and  R_LARCH_PCALA_LO12 are calculated in the same way.  Only take the lower 12 bits without other operations.  In earlier versions it was the same.

Reply:
This is an error and has been modified.